### ISSUE TITLE
Postinstallation script failure

### ISSUE THREAD

**Issue by rr4444** [bitnami/bitnami-docker-joomla#69]
Container post install script failing

Container log:
```

Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-joomla


Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-joomla/issues



nami    INFO  Initializing joomla


mysql-c INFO  Trying to connect to MySQL server


mysql-c INFO  Found MySQL server listening at mariadb:3306


mysql-c INFO  MySQL server listening and working at mariadb:3306


Error executing 'postInstallation': File /opt/bitnami/joomla/installation/sql/mysql/joomla.sql did not match any of the provided substitutions
```
Mariadb DB and tables made and DB user successfully generated. 

When restarting the container, I sometimes manage to catch the blue "Your application is being installed, please wait" Bitnami webpage, but after that browser unable to connect and no response/unavailable on the port that showed the Bitnami page

**Comment by rr4444** [comment]
I should also add that if I go into the bash console in the container, I am kicked out after a short time as the container keeps restarting trying to run that postinstall script

**Comment by rr4444** [comment]
If I rebuild the container from the image, I notice that the first run of the containter is the following

`
Welcome to the Bitnami joomla container


Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-joomla


Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-joomla/issues



nami    INFO  Initializing apache


nami    INFO  apache successfully initialized


nami    INFO  Initializing php


nami    INFO  php successfully initialized


nami    INFO  Initializing mysql-client


nami    INFO  mysql-client successfully initialized


nami    INFO  Initializing joomla


mysql-c INFO  Trying to connect to MySQL server


mysql-c INFO  Found MySQL server listening at mariadb:3306


mysql-c INFO  MySQL server listening and working at mariadb:3306


Error executing 'postInstallation': MySQL command failed to run. Error: 


 ERROR 1364 (HY000) at line 1: Field 'params' doesn't have a default value
`
Before repeating the above error in a loop

**Comment by rr4444** [comment]
Container is:

> {
>     "AppArmorProfile": "docker-default",
>     "Args": [
>         "httpd",
>         "-f",
>         "/opt/bitnami/apache/conf/httpd.conf",
>         "-DFOREGROUND"
>     ],
>     "Config": {
>         "AttachStderr": false,
>         "AttachStdin": false,
>         "AttachStdout": false,
>         "Cmd": [
>             "httpd",
>             "-f",
>             "/opt/bitnami/apache/conf/httpd.conf",
>             "-DFOREGROUND"
>         ],
>         "Domainname": "",
>         "Entrypoint": [
>             "/app-entrypoint.sh"
>         ],
>         "Env": [
>             "MARIADB_ROOT_PASSWORD=adminadmin",
>             "PATH=/opt/bitnami/apache/bin:/opt/bitnami/common/bin:/opt/bitnami/php/bin:/opt/bitnami/php/sbin:/opt/bitnami/mysql/bin:/opt/bitnami/nami/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
>             "ALLOW_EMPTY_PASSWORD=no",
>             "APACHE_ENABLE_CUSTOM_PORTS=no",
>             "APACHE_HTTPS_PORT_NUMBER=443",
>             "APACHE_HTTP_PORT_NUMBER=80",
>             "BITNAMI_APP_NAME=joomla",
>             "BITNAMI_IMAGE_VERSION=3.9.18-debian-10-r14",
>             "JOOMLA_DATABASE_NAME=joomla_site",
>             "JOOMLA_DATABASE_USER=joomla_site",
>             "JOOMLA_EMAIL=user@example.com",
>             "JOOMLA_FIRST_NAME=User",
>             "JOOMLA_LAST_NAME=Name",
>             "JOOMLA_PASSWORD=bitnami",
>             "JOOMLA_USERNAME=user",
>             "MARIADB_HOST=mariadb",
>             "MARIADB_PORT_NUMBER=3306",
>             "MARIADB_ROOT_USER=root",
>             "MYSQL_CLIENT_CREATE_DATABASE_PRIVILEGES=ALL",
>             "MYSQL_CLIENT_ENABLE_SSL=no",
>             "PHP_MEMORY_LIMIT=256M",
>             "SMTP_PROTOCOL=tls",
>             "JOOMLA_DATABASE_PASSWORD=joomla_site",
>             "MYSQL_CLIENT_CREATE_DATABASE_NAME=",
>             "MYSQL_CLIENT_CREATE_DATABASE_PASSWORD=",
>             "MYSQL_CLIENT_CREATE_DATABASE_USER=",
>             "MYSQL_CLIENT_SSL_CA_FILE=",
>             "SMTP_HOST=",
>             "SMTP_PASSWORD=",
>             "SMTP_PORT=",
>             "SMTP_USER="
>         ],
>         "ExposedPorts": {
>             "443/tcp": {},
>             "80/tcp": {},
>             "8181/tcp": {}
>         },
>         "Hostname": "d1b5455be50b",
>         "Image": "bitnami/joomla:latest",
>         "Labels": {
>             "maintainer": "Bitnami <containers@bitnami.com>"
>         },
>         "OnBuild": null,
>         "OpenStdin": false,
>         "StdinOnce": false,
>         "Tty": false,
>         "User": "",
>         "Volumes": {
>             "/bitnami": {}
>         },
>         "WorkingDir": ""
>     },
>     "Created": "2020-05-06T21:38:20.973578955Z",
>     "Driver": "overlay2",
>     "ExecIDs": null,
>     "GraphDriver": {
>         "Data": {
>             "LowerDir": "/var/lib/docker/overlay2/f3b38458cba9eca943e32393f9eccd20042b2effefacbd2cf141bf4188b7f8b9-init/diff:/var/lib/docker/overlay2/02691ef722f4c4bddbb8170a440717c033834560b103f85273e09b2e02e04390/diff:/var/lib/docker/overlay2/ecf3a5280cc75b52230bbd5461d82694347633b0213a82e638205ad043a40c3f/diff:/var/lib/docker/overlay2/8711892800f109df08777cdc8b2d5d93f4aa4d31a56fb8dbb9272e264cf1a413/diff:/var/lib/docker/overlay2/f2537f6f544c76244ea9870272d6312466e604254a3e5ead94df4079c30effcf/diff:/var/lib/docker/overlay2/0d6d600912b1704092becc5570d596e1bcaa9490be0fe581e2c5eb00987ce4e8/diff:/var/lib/docker/overlay2/5ccbe17257a73fd422ffe04d9df0f29f9db3ddca125638c533dc10a1711064f0/diff:/var/lib/docker/overlay2/83b2f3a9d245a92c00eedd989001c6818e66133dc1c91023da57d787c147b517/diff:/var/lib/docker/overlay2/97ac47aeec6e86097f88c07605f838362676b1343c6ba0ac7f1a756ec4935940/diff:/var/lib/docker/overlay2/12da6b46cbd7429f2f3599a8680bf080b41136c2a9be0d056ad6c96bd510c5d4/diff:/var/lib/docker/overlay2/3818d8c2f1f2325d81698584237ca3ac86354e887013793b256bada226d175d7/diff:/var/lib/docker/overlay2/627ac515df6299d7cad6e34c1760ff34d86ada576c211eb18b7b7a6f997f1741/diff:/var/lib/docker/overlay2/3bcd477477437b9735a3741d0ad5a6e9bb7faa3da26ca0b6574686eeb6bc83d9/diff:/var/lib/docker/overlay2/035ccf0287c0f387c5e90e55658ab64a5c019d054ef0a2fbd560a21107928b93/diff:/var/lib/docker/overlay2/d519639269c2ca3baa53c1fb1c3c1bcb1d83b5a070990e29552bba43037ed22d/diff:/var/lib/docker/overlay2/5bd15703c08642ac109c27c986f6db431399bbbf3c982e5e322d057e95504dea/diff",
>             "MergedDir": "/var/lib/docker/overlay2/f3b38458cba9eca943e32393f9eccd20042b2effefacbd2cf141bf4188b7f8b9/merged",
>             "UpperDir": "/var/lib/docker/overlay2/f3b38458cba9eca943e32393f9eccd20042b2effefacbd2cf141bf4188b7f8b9/diff",
>             "WorkDir": "/var/lib/docker/overlay2/f3b38458cba9eca943e32393f9eccd20042b2effefacbd2cf141bf4188b7f8b9/work"
>         },
>         "Name": "overlay2"
>     },
>     "HostConfig": {
>         "AutoRemove": false,
>         "Binds": [
>             "joomla-site:/bitnami"
>         ],
>         "BlkioDeviceReadBps": null,
>         "BlkioDeviceReadIOps": null,
>         "BlkioDeviceWriteBps": null,
>         "BlkioDeviceWriteIOps": null,
>         "BlkioWeight": 0,
>         "BlkioWeightDevice": null,
>         "CapAdd": [
>             "AUDIT_WRITE",
>             "CHOWN",
>             "DAC_OVERRIDE",
>             "FOWNER",
>             "FSETID",
>             "KILL",
>             "MKNOD",
>             "NET_BIND_SERVICE",
>             "NET_RAW",
>             "SETFCAP",
>             "SETGID",
>             "SETPCAP",
>             "SETUID",
>             "SYS_CHROOT"
>         ],
>         "CapDrop": [
>             "AUDIT_CONTROL",
>             "BLOCK_SUSPEND",
>             "DAC_READ_SEARCH",
>             "IPC_LOCK",
>             "IPC_OWNER",
>             "LEASE",
>             "LINUX_IMMUTABLE",
>             "MAC_ADMIN",
>             "MAC_OVERRIDE",
>             "NET_ADMIN",
>             "NET_BROADCAST",
>             "SYSLOG",
>             "SYS_ADMIN",
>             "SYS_BOOT",
>             "SYS_MODULE",
>             "SYS_NICE",
>             "SYS_PACCT",
>             "SYS_PTRACE",
>             "SYS_RAWIO",
>             "SYS_RESOURCE",
>             "SYS_TIME",
>             "SYS_TTY_CONFIG",
>             "WAKE_ALARM"
>         ],
>         "Capabilities": null,
>         "Cgroup": "",
>         "CgroupParent": "",
>         "ConsoleSize": [
>             0,
>             0
>         ],
>         "ContainerIDFile": "",
>         "CpuCount": 0,
>         "CpuPercent": 0,
>         "CpuPeriod": 0,
>         "CpuQuota": 0,
>         "CpuRealtimePeriod": 0,
>         "CpuRealtimeRuntime": 0,
>         "CpuShares": 0,
>         "CpusetCpus": "",
>         "CpusetMems": "",
>         "DeviceCgroupRules": null,
>         "DeviceRequests": null,
>         "Devices": [],
>         "Dns": [],
>         "DnsOptions": null,
>         "DnsSearch": null,
>         "ExtraHosts": [],
>         "GroupAdd": null,
>         "IOMaximumBandwidth": 0,
>         "IOMaximumIOps": 0,
>         "IpcMode": "private",
>         "Isolation": "",
>         "KernelMemory": 0,
>         "KernelMemoryTCP": 0,
>         "Links": null,
>         "LogConfig": {
>             "Config": {},
>             "Type": "json-file"
>         },
>         "MaskedPaths": [
>             "/proc/asound",
>             "/proc/acpi",
>             "/proc/kcore",
>             "/proc/keys",
>             "/proc/latency_stats",
>             "/proc/timer_list",
>             "/proc/timer_stats",
>             "/proc/sched_debug",
>             "/proc/scsi",
>             "/sys/firmware"
>         ],
>         "Memory": 0,
>         "MemoryReservation": 0,
>         "MemorySwap": 0,
>         "MemorySwappiness": null,
>         "NanoCpus": 0,
>         "NetworkMode": "joomla-site",
>         "OomKillDisable": false,
>         "OomScoreAdj": 0,
>         "PidMode": "",
>         "PidsLimit": null,
>         "PortBindings": {
>             "80/tcp": [
>                 {
>                     "HostIp": "",
>                     "HostPort": "9003"
>                 }
>             ]
>         },
>         "Privileged": false,
>         "PublishAllPorts": false,
>         "ReadonlyPaths": [
>             "/proc/bus",
>             "/proc/fs",
>             "/proc/irq",
>             "/proc/sys",
>             "/proc/sysrq-trigger"
>         ],
>         "ReadonlyRootfs": false,
>         "RestartPolicy": {
>             "MaximumRetryCount": 0,
>             "Name": "always"
>         },
>         "Runtime": "runc",
>         "SecurityOpt": null,
>         "ShmSize": 67108864,
>         "UTSMode": "",
>         "Ulimits": null,
>         "UsernsMode": "",
>         "VolumeDriver": "",
>         "VolumesFrom": null
>     },
>     "HostnamePath": "/var/lib/docker/containers/39b33163dd118fb78090c0e43062ba61db86c8f28f77c4120983ec108c2052d9/hostname",
>     "HostsPath": "/var/lib/docker/containers/39b33163dd118fb78090c0e43062ba61db86c8f28f77c4120983ec108c2052d9/hosts",
>     "Id": "39b33163dd118fb78090c0e43062ba61db86c8f28f77c4120983ec108c2052d9",
>     "Image": "sha256:849af151c1b494886673f869d30c8fae7347d14d740b611d3d4b86801a172af3",
>     "LogPath": "/var/lib/docker/containers/39b33163dd118fb78090c0e43062ba61db86c8f28f77c4120983ec108c2052d9/39b33163dd118fb78090c0e43062ba61db86c8f28f77c4120983ec108c2052d9-json.log",
>     "MountLabel": "",
>     "Mounts": [
>         {
>             "Destination": "/bitnami",
>             "Driver": "local",
>             "Mode": "z",
>             "Name": "joomla-site",
>             "Propagation": "",
>             "RW": true,
>             "Source": "/var/lib/docker/volumes/joomla-site/_data",
>             "Type": "volume"
>         }
>     ],
>     "Name": "/bitnami_joomla-site",
>     "NetworkSettings": {
>         "Bridge": "",
>         "EndpointID": "",
>         "Gateway": "",
>         "GlobalIPv6Address": "",
>         "GlobalIPv6PrefixLen": 0,
>         "HairpinMode": false,
>         "IPAddress": "",
>         "IPPrefixLen": 0,
>         "IPv6Gateway": "",
>         "LinkLocalIPv6Address": "",
>         "LinkLocalIPv6PrefixLen": 0,
>         "MacAddress": "",
>         "Networks": {
>             "joomla-site": {
>                 "Aliases": [
>                     "f87100e15a94",
>                     "033a3ff97067",
>                     "95b67d6953a9",
>                     "39b33163dd11"
>                 ],
>                 "DriverOpts": null,
>                 "EndpointID": "",
>                 "Gateway": "",
>                 "GlobalIPv6Address": "",
>                 "GlobalIPv6PrefixLen": 0,
>                 "IPAMConfig": {},
>                 "IPAddress": "",
>                 "IPPrefixLen": 0,
>                 "IPv6Gateway": "",
>                 "Links": null,
>                 "MacAddress": "",
>                 "NetworkID": "5150902655a57e412e8078db2430ccbc505a641d4845d10091623bcd5f2899c2"
>             }
>         },
>         "Ports": {},
>         "SandboxID": "310c52e90f2a7202f28e293eed748211208a100638efa0a5f946759b66e8f58f",
>         "SandboxKey": "/var/run/docker/netns/310c52e90f2a",
>         "SecondaryIPAddresses": null,
>         "SecondaryIPv6Addresses": null
>     },
>     "Path": "/app-entrypoint.sh",
>     "Platform": "linux",
>     "Portainer": {
>         "ResourceControl": {
>             "Id": 59,
>             "ResourceId": "39b33163dd118fb78090c0e43062ba61db86c8f28f77c4120983ec108c2052d9",
>             "SubResourceIds": [],
>             "Type": 1,
>             "UserAccesses": [],
>             "TeamAccesses": [],
>             "Public": false,
>             "AdministratorsOnly": true,
>             "System": false
>         }
>     },
>     "ProcessLabel": "",
>     "ResolvConfPath": "/var/lib/docker/containers/39b33163dd118fb78090c0e43062ba61db86c8f28f77c4120983ec108c2052d9/resolv.conf",
>     "RestartCount": 3,
>     "State": {
>         "Dead": false,
>         "Error": "",
>         "ExitCode": 1,
>         "FinishedAt": "2020-05-06T21:41:38.515418063Z",
>         "OOMKilled": false,
>         "Paused": false,
>         "Pid": 0,
>         "Restarting": false,
>         "Running": false,
>         "StartedAt": "2020-05-06T21:39:12.907221405Z",
>         "Status": "exited"
>     }
> }

**Comment by marcosbc** [comment]
Hi @rr4444, could you share the docker-compose.yml file you used for deploying the image? Or the "docker run" command if it were the case.

It could be possible you're using an external database, and for there to be a version mismatch. The container is expecting MariaDB 10.1 so if you have any other version please let us know.

**Comment by rr4444** [comment]
Hi there, thanks, I just checked the version. I am using your latest bitnami mariadb as an external, but that is running mariadb 10.3.22-debian-10-r108. the joomla image was also the latest

I have launched this via Portainer, so don't have the run command it issues in the background, but the output in my last post was from inspecting the container (hopefully that suffices)

**Comment by rr4444** [comment]
i was attracted to bitnami because I was hoping versioning and other such things between dependent apps would be consistent

but I am surprised that it would matter that much, if it is that. is there some way that version check can be bypassed/updated?

thanks again

**Comment by marcosbc** [comment]
I see. The problem is that Joomla itself does not support MariaDB 10.3 (hence the errors importing joomla.sql). If you stick with MariaDB 10.1 it should work fine.

If you find issues with MariaDB 10.1 let us know. I'll close the case, but you'll still be able to comment if needed.

**Comment by rr4444** [comment]
Arg... 10.1 is end of life in October, 5 months away...!

**Comment by rr4444** [comment]
I am very sure 10.1 isn't the requirement tbh...

**Comment by rr4444** [comment]
From what I can see mariadb 10.3 is equivalent to mysql 5.7, which is fine for Joomla in my experience, here too https://forum.joomla.org/viewtopic.php?f=710&t=966030

It;s mysql 8 that is a joomla problem

Can the script be updated?

**Comment by rr4444** [comment]
Official joomla build working with your mariadb 10.3 docker image :)

**Comment by marcosbc** [comment]
Thanks for letting us know. In that case we will investigate further.

The container currently relies on `/opt/bitnami/joomla/installation/sql/mysql/joomla.sql` which seems incompatible with MariaDB 10.3, but it is possible they're using another automation for initializing the database.

We will get back to you once we have more information.

**Comment by rr4444** [comment]
Thanks, appreciated. Don't want bitnami joomla to die in less than six months :(

**Comment by marcosbc** [comment]
Hi @rr4444, after more investigation I could trace these issue back to Joomla not supporting the MySQL/MariaDB `STRICT_TRANS_TABLES` option. The following related errors are shown:

* `ERROR 1364 (HY000): Field 'params' doesn't have a default value`
* `Out of range value for column 'registerDate' at row 1`
* `ERROR 1292 (22007): Incorrect datetime value: '0000-00-00 00:00:00' for column 'registerDate' at row 1`

It is very likely that when you launched the official Joomla container with MySQL 8+ or MariaDB 10.2+, the `sql_mode` did not contain that option (`SELECT @@sql_mode`).

We'll look into finding a way so you can specify which sql_mode options you want to enable during the MySQL/MariaDB database initialization, which would allow you to workaround this issue until Joomla starts officially supporting MySQL 8.0+/MariaDB 10.2+.

**Comment by rr4444** [comment]
I am not saying that you are wrong, as I am missing a link in my head, but please note that I have official joomla image working with the bitnami mariadb 10.3 with zero problems

**Comment by marcosbc** [comment]
That is probably the case because Joomla does not pre-configure a user, something the Bitnami Joomla Docker image does, since the issue happens during the initialization phase when setting up the user in the database (not using the Joomla web wizard).

**Comment by marcosbc** [comment]
Hi,

We just added support for setting a custom sql_mode option for MySQL, hence supporting MySQL 5.7+ and MariaDB 10.2+. Find an example of the changes to perform below:

```
services:
  ...
  mysql:
    image: 'bitnami/mysql:8.0'
    ports:
      - '3306:3306'
    volumes:
      - 'mysql_data:/bitnami/mysql/data'
    environment:
      - MYSQL_USER=bn_joomla
      - MYSQL_DATABASE=bitnami_joomla
      - ALLOW_EMPTY_PASSWORD=yes
      - MYSQL_SQL_MODE=NO_ZERO_IN_DATE,NO_ZERO_DATE,NO_ENGINE_SUBSTITUTION
volumes:
  ...
  mysql_data:
    driver: local
```

**Comment by marcosbc** [comment]
NOTE: It is important you ensure you're using the latest `bitnami/mysql:8.0` image, as the changes were added recently (`8.0.20-debian-10-r37`), i.e.:

```
$ docker pull bitnami/mysql:8.0
```

**Comment by marcosbc** [comment]
I forgot to mention that the MySQL default_authentication_plugin environment needs to be added as well (for MySQL 8.0 only). In summary, you would need the following configuration:

* MySQL 5.7, MariaDB 10.2+: `MYSQL_SQL_MODE=...`.
* MySQL 8.0: `MYSQL_SQL_MODE=...` and `MYSQL_AUTHENTICATION_PLUGIN=mysql_native_password`.

**Comment by rr4444** [comment]
Sorry, am a bit confused. Does this now require Mysql 8? 

> It;s mysql 8 that is a joomla problem

Used to be the case that joomla does not work with mysql 8. I think that's still the case...

**Comment by marcosbc** [comment]
> Sorry, am a bit confused. Does this now require Mysql 8?

No, feel free to use MariaDB:

```
services:
  ...
  mariadb:
    image: 'docker.io/bitnami/mariadb:10.4-debian-10'
    volumes:
      - 'mariadb_data:/bitnami'
    environment:
      - MARIADB_USER=bn_joomla
      - MARIADB_DATABASE=bitnami_joomla
      - ALLOW_EMPTY_PASSWORD=yes
      - MARIADB_SQL_MODE=NO_ZERO_IN_DATE,NO_ZERO_DATE,NO_ENGINE_SUBSTITUTION
volumes:
  ...
  mariadb_data:
    driver: local
```

> Used to be the case that joomla does not work with mysql 8. I think that's still the case...

We've been using MySQL 8 in the [Bitnami Joomla Stack](https://bitnami.com/stack/joomla) without problems, but note that we configured MySQL to contain a supported value for `sql_mode` and `mysql_authentication_plugin`.

**Comment by rr4444** [comment]
Hi, joys of covid... :(

I still have the same problem, with logs looping this container restart:

```
Welcome to the Bitnami joomla container


Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-joomla


Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-joomla/issues



nami    INFO  Initializing joomla


mysql-c INFO  Trying to connect to MySQL server


mysql-c INFO  Found MySQL server listening at mariadb:3306


mysql-c INFO  MySQL server listening and working at mariadb:3306


Error executing 'postInstallation': File /opt/bitnami/joomla/installation/sql/mysql/joomla.sql did not match any of the provided substitutions



Welcome to the Bitnami joomla container


Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-joomla


Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-joomla/issues



nami    INFO  Initializing joomla


mysql-c INFO  Trying to connect to MySQL server


mysql-c INFO  Found MySQL server listening at mariadb:3306


mysql-c INFO  MySQL server listening and working at mariadb:3306


Error executing 'postInstallation': File /opt/bitnami/joomla/installation/sql/mysql/joomla.sql did not match any of the provided substitutions

```
Maria DB compose is:

```
 mariadb:
    image: bitnami/mariadb:latest
    volumes:
      - 'mariadb:/bitnami/mariadb'
    networks: 
      - mariadb
    restart: unless-stopped
    expose:
      - 3306/tcp
    environment:
      - MARIADB_ROOT_PASSWORD=blahblah
      # Added for Joomla support in Mysql5.7 equivs and up based on https://github.com/bitnami/bitnami-docker-joomla/issues/69#issuecomment-641241602
      - MARIADB_SQL_MODE=NO_ZERO_IN_DATE,NO_ZERO_DATE,NO_ENGINE_SUBSTITUTION
```

Joomla compose is:

```
joomla:
    image: 'bitnami/joomla'
    volumes:
      - 'joomla:/var/www/html'
    networks: 
      - web
      - mariadb
    restart: unless-stopped
    environment:
      - MARIADB_HOST=mariadb
      - JOOMLA_DATABASE_USER=blah
      - JOOMLA_DATABASE_PASSWORD=blah
      - JOOMLA_DATABASE_NAME=blah
      - MARIADB_SQL_MODE=NO_ZERO_IN_DATE,NO_ZERO_DATE,NO_ENGINE_SUBSTITUTION
  ```

**Comment by rr4444** [comment]
I even added the ENV MARIADB_SQL_MODE in both 'for fun' as I am getting a bit desperate with this...

**Comment by rr4444** [comment]
I hope covid is treating you better than me. Any additional help to get this going will be much appreciated.

**Comment by rr4444** [comment]
It must be todo with the mariadb authentication issues you mentioned on the mysql front, as the mariadb logs say:

`2020-07-06 12:12:46 74 [Warning] Aborted connection 74 to db: 'unconnected' user: 'unauthenticated' host: '>MARIADBIP>' (This connection closed normally without authentication)`

I should add that bitnami mariadb latest seems to be v10.5 now, if that makes any difference. 

This mariadb instance is also used by other things, so I am worried about killing everything else's authentication. Joomla itself is fine underneath, it's the bitnami install script that seems to need fixing

**Comment by rr4444** [comment]
Also tried the mysql8 command param to the compose:

`--default-authentication-plugin=mysql_native_password`

just to see.. but it made no difference. And i saw a req for this param on mariadb github, so it doesn't exist and i was being desperate again...

**Comment by rr4444** [comment]
I've tried deleting post init scripts to try to stop the issue, with no success. brute force desperation, i know. 

I think I am being silly persisting and will now just move to the apache image and put on joomla manually. this doesn't work in a joomla:mariadb bitnami image combo, very sadly :(

**Comment by rr4444** [comment]
Famous last words, the apache image has no php and is not run as root, so i can't install it

**Comment by marcosbc** [comment]
Hi @rr4444, I just tried with the following `docker-compose.yml` and it worked without issues:

```
version: '2'
services:
  mariadb:
    image: 'docker.io/bitnami/mariadb:latest'
    environment:
      - MARIADB_USER=bn_joomla
      - MARIADB_DATABASE=bitnami_joomla
      - ALLOW_EMPTY_PASSWORD=yes
      - MARIADB_SQL_MODE=NO_ENGINE_SUBSTITUTION
    volumes:
      - 'mariadb_data:/bitnami'
  joomla:
    image: 'docker.io/bitnami/joomla:latest'
    environment:
      - MARIADB_HOST=mariadb
      - MARIADB_PORT_NUMBER=3306
      - JOOMLA_DATABASE_USER=bn_joomla
      - JOOMLA_DATABASE_NAME=bitnami_joomla
      - ALLOW_EMPTY_PASSWORD=yes
    ports:
      - '80:80'
      - '443:443'
    volumes:
      - 'joomla_data:/bitnami'
    depends_on:
      - mariadb
volumes:
  mariadb_data:
    driver: local
  joomla_data:
    driver: local
```

Could you make sure to remove the volumes before deploying again? Maybe that does the trick.  Also, make sure to use the proper image/tag for the Joomla and MariaDB images, to avoid issues.

NOTE: In the example above I used the `latest` tag just as an example. [In production you should avoid rolling tags](https://docs.bitnami.com/tutorials/understand-rolling-tags-containers/).

**Comment by rr4444** [comment]
> Could you make sure to remove the volumes before deploying again? Maybe that does the trick. Also, make sure to use the proper image/tag for the Joomla and MariaDB images, to avoid issues with outdated images.

I did this, but then it gave me another random idea to check. I changed the name of the service in the compose file. That did the trick. I do not understand why. The only thing that I can thing of is that between finding the original problem and retrying with the new env vars I moved cloud server. I rsynced -zarvh /var/lib/docker, which take permissions etc. Only thing I can think of.... everything is happy. 

Many thanks. If you hadn't suggested deleting volumes I wouldn't have thought of changing the service name.
 
> NOTE: In the example above I used the `latest` tag just as an example. [In production you should avoid rolling tags](https://docs.bitnami.com/tutorials/understand-rolling-tags-containers/).

I know those, but totally forgot. Which means I would have kicked myself in disappointment even harder if it had caused me problems. Thank you for highlighting. 

I am very pleased to be able to use your images. They really are a critical part of the ecosystem. Thanks again
